luskan/src/test/java/org/opencadc/luskan/SiteLocationsConverterTest.java

```diff
@@ -0,0 +1,171 @@
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.luskan;
+
+import ca.nrc.cadc.tap.TapQuery;
+import ca.nrc.cadc.tap.schema.TapSchema;
+import ca.nrc.cadc.util.Log4jInit;
+import ca.nrc.cadc.util.MultiValuedProperties;
+import ca.nrc.cadc.uws.Parameter;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class SiteLocationsConverterTest {
+    private static final Logger log = Logger.getLogger(SiteLocationsConverterTest.class);
+
+    static {
+        Log4jInit.setLevel("org.opencadc.luskan", Level.INFO);
+    }
+
+    private static final TapSchema tapSchema = TestUtil.loadTapSchema();
+
+    private static final String selectList =
+        "inventory.artifact.uri, inventory.artifact.uribucket, inventory.artifact.contentchecksum, "
+            + "inventory.artifact.contentLastmodified, inventory.artifact.contentlength, "
+            + "inventory.artifact.contenttype, inventory.artifact.contentencoding, inventory.artifact.lastmodified, "
+            + "inventory.artifact.metachecksum, inventory.artifact.id ";
+    private static final String fromClause = "FROM inventory.artifact";
+    private static final String isNotNullConstraint = "(inventory.artifact.sitelocations IS NOT NULL)";
+
+    @Test
+    public void testSelectAll() {
+        String query = "select * from inventory.artifact";
+        String expected = "SELECT " + selectList + fromClause;
+        doTest(query, expected, null);
+    }
+
+    @Test
+    public void testSelectColumns() {
+        String query = "select uri, lastmodified from inventory.artifact";
+        String expected = "SELECT uri, lastModified " + fromClause + " WHERE " + isNotNullConstraint;
+        doTest(query, expected, "true");
+    }
+
+    @Test
+    public void testSelectColumnWithAlias() {
+        String query = "select uri as \"foo\" from inventory.artifact";
+        String expected = "SELECT uri as \"foo\" " + fromClause + " WHERE " + isNotNullConstraint;
+        doTest(query, expected, "true");
+    }
+
+    @Test
+    public void testSelectWithWhere() {
+        String query = "select uri from inventory.artifact where uribucket = 1";
```

Comment:
this test is fine except that uriBucket is a char column so the value should have single ticks. It doesn't matter for the purposes of query mangling but it's distracting... maybe `contentLength <= 1024` would be less distracting.

Reply:
Updated the test query as above.